src/chunk.c

```diff
@@ -4551,3 +4551,68 @@ ts_chunk_attach_osm_table_chunk(PG_FUNCTION_ARGS)
 
 	PG_RETURN_BOOL(ret);
 }
+
+static ScanTupleResult
+chunk_tuple_osm_chunk_found(TupleInfo *ti, void *arg)
+{
+	bool isnull;
+	Datum osm_chunk = slot_getattr(ti->slot, Anum_chunk_osm_chunk, &isnull);
+
+	Assert(!isnull);
+	bool is_osm_chunk = DatumGetBool(osm_chunk);
+
+	if (!is_osm_chunk)
+		return SCAN_CONTINUE;
+
+	int *chunk_id = (int *) arg;
+	Datum chunk_id_datum = slot_getattr(ti->slot, Anum_chunk_id, &isnull);
+	Assert(!isnull);
+	*chunk_id = DatumGetInt32(chunk_id_datum);
+	return SCAN_DONE;
+}
+
+/* get OSM chunk id associated with the hypertable */
+int
+ts_chunk_get_osm_chunk_id(int hypertable_id)
+{
+	int chunk_id = INVALID_CHUNK_ID;
+	ScanKeyData scankey[2];
+	bool is_osm_chunk = true;
+	Catalog *catalog = ts_catalog_get();
+	ScannerCtx scanctx = {
+		.table = catalog_get_table_id(catalog, CHUNK),
+		.index = catalog_get_index(catalog, CHUNK, CHUNK_OSM_CHUNK_INDEX),
+		.nkeys = 2,
+		.scankey = scankey,
+		.data = &chunk_id,
+		.tuple_found = chunk_tuple_osm_chunk_found,
+		.lockmode = AccessShareLock,
+		.scandirection = ForwardScanDirection,
+		//.result_mctx = mctx,
+	};
+
+	/*
+	 * Perform an index scan on hypertable ID + osm_chunk
+	 */
+	ScanKeyInit(&scankey[0],
+				Anum_chunk_osm_chunk_idx_osm_chunk,
+				BTEqualStrategyNumber,
+				F_BOOLEQ,
+				BoolGetDatum(is_osm_chunk));
+	ScanKeyInit(&scankey[1],
+				Anum_chunk_osm_chunk_idx_hypertable_id,
+				BTEqualStrategyNumber,
+				F_INT4EQ,
+				Int32GetDatum(hypertable_id));
+
+	int num_found = ts_scanner_scan(&scanctx);
+
+	if (num_found > 1)
+	{
+		ereport(ERROR,
+				(errcode(ERRCODE_TS_INTERNAL_ERROR),
+				 errmsg("More than 1 OSM chunk found for hypertable (%d)", hypertable_id)));
```

Comment:
Can we test it?

Reply:
this is unexpected corruption. can test only by modifying catalog entries manually.